zone/mob_ai.cpp

```diff
@@ -1762,6 +1768,10 @@ void Mob::AI_Event_Engaged(Mob* attacker, bool iYellForHelp) {
 
 	SetAppearance(eaStanding);
 
+	// Idle timer is huge, gives NPCs an early disadvantage when 1st entering
+	// combat.
+	this->CastToNPC()->AI_RestartSpellTimer();
```

Comment:
Just do the timer inline here instead of using a function, it doesn't have any params anyways

Reply:
Make it an inline function in the header? @Akkadius

Reply:
Sorry, not experienced with inline - seen it in headers ,,,  Would want it in header in case needs to be called elsewhere?

Reply:
Oh you mean just use AIautocastspell_timer->Start(300, false); inline.

I think we want the logging as it happens for mobs?  Also since it is already a function for mobs - seems more consistent?

Also the function it is being called in does not have access to the timer if I recall, without changes.

Reply:
You don't need to CastToNPC() because you're already in NPC. Since this is a NPC class timer - you can call AIautocastspell_timer->Start(300, false); inline where the function call was made

Reply:
This function is mob:: not npc.  I think it is not accessable.

Reply:
![image](https://user-images.githubusercontent.com/3319450/34910145-a9b57660-f874-11e7-8e05-5a51981d2698.png)

It is confusing with the diff - an AI function isn't typically 'Mob' you can CastToNPC inside the mob function with an if NPC check before the timer. This code needs to be cleaned up in general